pytorch_lightning/plugins/checkpoint/torch.py

```diff
@@ -0,0 +1,57 @@
+# Copyright The PyTorch Lightning team.
```

Comment:
Similarly, rename the file to torch_checkpoint_io_plugin.py?

or even better

pytorch_lightning/plugins/io/torch_plugin.py

In any case I'd avoid `torch.py` as the filename which could be conflicting with importing torch itself.